### PR TITLE
improvement: support relabelings in ServiceMonitor

### PR DESCRIPTION
**Pull request by scoof**
**Describe your changes**
Added optional support for metricRelabelings

**Testing performed**
ServiceMonitor renders cleanly with or without relabelings

### CODE REVIEW

**Comment by hcsaustrup** [comment]
Prioritize this, please.

File: charts/goldpinger/values.yaml [hunk 1]
```diff
@@ -143,6 +143,10 @@ serviceMonitor:
   # namespace: monitoring
   interval: 30s
   # honorLabels: true
+  metricRelabelings: {}
```
Comment (skamboj) [inline charts/goldpinger/values.yaml:146]: My main concern is the `metricsRelabelings` should be a list and I am not sure I like the part where we are setting it to an empty object `{}` here - I would rather set this to an empty list `[]`? 

(Even though your example below kind of makes it clear we are expecting a list... I still think the type is important here)

**Review by skamboj — COMMENTED** [review]
Thank you so much for your contribution and apologies for the delay in getting to this.

I have a small comment about changing the default type of `metricsRelabelings` but other than that, this looks good.

File: charts/goldpinger/values.yaml (hunk 1 above)
Comment (scoof) [reply charts/goldpinger/values.yaml]: That's fair. Done.

**Review by skamboj — APPROVED** [review]
Thank you so much for this and sorry for the delay in getting to it. LGTM

:shipit:

**Pull request merged**